deepchecks/utils/metrics.py

```diff
@@ -109,6 +110,10 @@ def filter_nulls(cls, dataset: 'tabular.Dataset') -> 'tabular.Dataset':
         valid_idx = dataset.data[dataset.label_name].notna()
         return dataset.copy(dataset.data[valid_idx])
 
+    def run_on_data_and_label(self, model, data: pd.DataFrame, label_col):
+        """Run scorer with model, data and labels without null filtering."""
+        return self.scorer(model, data, label_col)
+
```

Comment:
Why not just change the __call__ function to have the null filtering as optional?

Reply:
Also, why not use _run_score?

Reply:
The problem is not the null filtering but the fact that creating a dataset per leaf data just to send to this function is slow